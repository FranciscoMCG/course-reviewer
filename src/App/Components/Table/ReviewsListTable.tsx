import React from "react";
import { Review } from "../../types/types";

const headers = ["First Name", "Last Name", "Email", "Course", "Review"];

const ReviewsListTable = ({ reviews }: any) => (
  <div className="table-wrapper">
    <table className="fl-table">
      <thead>
        <tr>
          {headers.map((header, index) => (
            <th key={index}>{header}</th>
          ))}
        </tr>
      </thead>
      <tbody>
        {reviews &&
          reviews.map((review: Review) => (
            <tr key={review._id}>
              <td>{review && review.firstName}</td>
              <td>{review && review.lastName}</td>
              <td>{review && review.email}</td>
              <td>{review && review.course}</td>
              <td>{review && review.review}</td>
            </tr>
          ))}
      </tbody>
    </table>
  </div>
);

export default ReviewsListTable;
