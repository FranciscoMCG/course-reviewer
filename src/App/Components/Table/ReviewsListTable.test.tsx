import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

import ReviewsListTable from "./ReviewsListTable";

describe("<ReviewsListTable />", () => {
  it("should render component", () => {
    const wrapper = shallow(<ReviewsListTable />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it("renders empty message as table cell if there is no data", () => {
    const headers = [
      { header: "First Name" },
      { header: "Last Name" },
      { header: "Email" },
      { header: "Course" },
      { header: "Review" }
    ];

    const container = shallow(<ReviewsListTable reviews={[]} />);
    const table = container.find("table");
    expect(table).toHaveLength(1);
    const thead = table.find("thead");
    expect(thead).toHaveLength(1);
    const tableHeaders = thead.find("th");
    expect(tableHeaders).toHaveLength(headers.length);
    const tbody = table.find("tbody");
    expect(tbody).toHaveLength(1);
  });
});
