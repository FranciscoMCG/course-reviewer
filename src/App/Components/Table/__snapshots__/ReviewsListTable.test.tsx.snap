// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewsListTable /> should render component 1`] = `
<div
  className="table-wrapper"
>
  <table
    className="fl-table"
  >
    <thead>
      <tr>
        <th
          key="0"
        >
          First Name
        </th>
        <th
          key="1"
        >
          Last Name
        </th>
        <th
          key="2"
        >
          Email
        </th>
        <th
          key="3"
        >
          Course
        </th>
        <th
          key="4"
        >
          Review
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
